x-pack/plugins/endpoint/public/applications/endpoint/store/alerts/reducer.ts

```diff
@@ -27,6 +28,11 @@ export const alertListReducer: Reducer<AlertListState, AppAction> = (
       ...state,
       ...action.payload,
     };
+  } else if (action.type === 'userChangedUrl') {
```

Comment:
you are updating the Alerts store every time the URL changes - is that intentional? or did you forget to use `isOnAlertsList()`?

Also, when the user leaves the alert list page, the store for it should be reset.

Reply:
i suppose, only time the action is dispatched currently is if the app is already confirmed to be on the alerts page

Reply:
The action is dispatched anytime the URL changes - so navigating to `/management` or `/policy` will trigger the action.

Reply:
@paul-tavares We didn't want to force other teams to use this approach so we are storing the `location` in the alerts piece of the store. If all of us want to use this, we can store the location at the top level of the store.

Reply:
I created an issue for clearing the state when navigating away https://github.com/elastic/endpoint-app-team/issues/183

Reply:
@peluja1012 I actually thought we had agreed to use this new approach instead of `usePageId()` when we talked through the revised implementation around using `location` provided by react-router. At least I was planning on refactoring the Policy/Management pages to use `userChangedUrl` instead once this merged in. 

Not sure if storing it globally would facilitate things though - since currently we're building our reducers/middleware to NOT be aware of global state. Accessing the location information would be an issue thus why each concern needs to store it, no?

Reply:
@paul-tavares Oh that's nice. I didn't realize we were all moving forward with this approach. It would certainly be easier the way the code is laid out now to have each concern store the location. I don't know if I like the fact that we'll need a copied and pasted reducer clause to update the location in each concern though. I would rather have location be global. We could refactor the code to allow for a global location piece of state or create a reducer helper to handle updating the location in each concern. Either way, I'd like to do this work in a separate PR. This PR is currently blocking work on Alert Details and Alerts Filtering, so I would like to get it merged as soon as we can.

Reply:
agreed 👍